server/tserver/src/main/java/org/apache/accumulo/tserver/tablet/Tablet.java

```diff
@@ -1419,7 +1419,11 @@ private void closeConsistencyCheck() {
         }
       });
 
-      compareToDataInMemory(tabletMeta);
+      if (!tabletMeta.getFilesMap().equals(getDatafileManager().getDatafileSizes())) {
+        String msg = "Data files in " + extent + " differ from in-memory data "
+            + tabletMeta.getFilesMap() + " " + getDatafileManager().getDatafileSizes();
+        log.error(msg);
+      }
```

Comment:
Could these maps change between the check and the message construction?

```suggestion
      var filesMap = tabletMeta.getFilesMap();
      var datafileSizes = getDatafileManager().getDatafileSizes();
      if (!filesMap.equals(datafileSizes)) {
        String msg = "Data files in " + extent + " differ from in-memory data " + filesMap + " " + datafileSizes;
        log.error(msg);
      }
```

If the objects themselves are mutable, then it could be even more of a problem and we might need to do a protective copy to do the comparison.

Reply:
> Could these maps change between the check and the message construction?

No, neither one of them should change.

What `tabletMeta.getFilesMap()` returns comes originally from Ample when it read from the metadata table.  If this were changing it would be a really weird bug that would have a much wider impact.  What `getDatafileManager().getDatafileSizes()` returns is a copy of its files made using the same lock this method is holding.  So it should not change given the current code, but it would probably be cleaner to only get it once.

Reply:
I made an update re this in 14cf0d2c80c5ba55d54d25b1b2fa221a6f13dee8.  In that update I also adjusted the log message.